compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirFunctionParameterChecker.kt

```diff
@@ -70,4 +76,30 @@ object FirFunctionParameterChecker : FirFunctionChecker() {
             }
         }
     }
+
+    private fun checkUninitializedParameter(function: FirFunction<*>, context: CheckerContext, reporter: DiagnosticReporter) {
+        for ((index, parameter) in function.valueParameters.withIndex()) {
+            // Alas, CheckerContext.qualifiedAccesses stack is not available at this point.
+            // Thus, manually visit default value expression and report the diagnostic on qualified accesses of interest.
+            parameter.defaultValue?.accept(object : FirVisitorVoid() {
+                override fun visitElement(element: FirElement) {
+                    element.acceptChildren(this)
+                }
+
+                override fun visitQualifiedAccessExpression(qualifiedAccessExpression: FirQualifiedAccessExpression) {
+                    val namedReference = qualifiedAccessExpression.calleeReference as? FirResolvedNamedReference ?: return
+                    val referredParameter = namedReference.resolvedSymbol.fir as? FirValueParameter ?: return
+                    val referredParameterIndex = function.valueParameters.indexOf(referredParameter)
+                    if (index <= referredParameterIndex) {
+                        reporter.reportOnWithSuppression(
+                            qualifiedAccessExpression,
+                            FirErrors.UNINITIALIZED_PARAMETER,
+                            function.valueParameters[referredParameterIndex].symbol,
```

Comment:
Nit: Can just be `referredParameter.symbol`

Reply:
Duh 🤦